### PR TITLE
[Feat]: Centralised commmand registration

### PR DESCRIPTION
**Pull request by yatendra2001**
## Description

Added 3 new functionalities:

1. Created a centralised utility file to create and register commands.
2. Added telemetry for commands.
3. Added shortcut "cmd+alt+o" to toggle chat.
4. Made focus auto-enabled for chat input whenever the chat's webview is visible.

## Screen Recording
Area

### CODE REVIEW

File: media/chat/scripts/main.js [hunk 1]
```diff
@@ -357,8 +357,13 @@ class Mentionify {
                         fileNameContainer.appendChild(divBlock);
                     });
                 }
-                break;
             }
+            case 'focusChatInput':
+                {
+                    const chatInput = document.getElementById('prompt-input'); // Replace with your actual input element's ID
```
Comment (doitwithdk) [inline media/chat/scripts/main.js]: `prompt-input` is the actual input element Id isn't it?

File: package.json [hunk 2]
```diff
@@ -71,6 +71,11 @@
         "key": "ctrl+R",
         "mac": "cmd+R",
         "when": "editorHasSelection && resourceLangId == 'dart'"
+      },
+      {
+        "command": "fluttergpt.toggleChat",
+        "key": "ctrl+at+o",
```
Comment (doitwithdk) [inline package.json]: Spell check

**Review by doitwithdk — CHANGES_REQUESTED** [review]
LGTM other than the minor issues I mentioned. Good work!

File: media/chat/scripts/main.js (hunk 1 above)
Comment (yatendra2001) [reply media/chat/scripts/main.js]: correct `prompt-input` is the id that we use for our chatbox `textarea`
removing the unnecessary comment.

**Comment by yatendra2001** [comment]
@doitwithdk issues resolved.

**Review by doitwithdk — APPROVED** [review]

**Pull request merged**